tests/includes/aws_ami_creation.sh

```diff
@@ -29,11 +25,30 @@ launch_and_wait_ec2() {
 }
 
 create_ami_and_wait_available() {
-	local instance_id_ami_builder ami_id_result
+	local ami_id_result
+	ami_id_result=${1}
+
+	# Retrieve the image_id corresponding to ubuntu jammy
+	OUT=$(aws ec2 describe-images \
+		--owners 099720109477 \
+		--filters "Name=name,Values=ubuntu/images/hvm-ssd/ubuntu-jammy-?????-${MODEL_ARCH:-amd64}-server-????????" 'Name=state,Values=available' \
+		--query 'reverse(sort_by(Images, &CreationDate))[:1].ImageId' \
```

Comment:
SO, can you do some cleanup for me?

Can we add `MODEL_ARCH` to the main.sh file at the top. And export it.

```sh
export MODEL_ARCH=${MODEL_ARCH:-}
```

That way we don't get errors when running with verbose mode on. Then you'll need to change your thing above to an if statement.

```sh
arch="amd64"
if [[ -n "${MODEL_ARCH}" ]]; then
     arch="${MODEL_ARCH}"
fi

OUT=$(aws ec2 describe-images \
		--owners 099720109477 \
		--filters "Name=name,Values=ubuntu/images/hvm-ssd/ubuntu-jammy-?????-${arch}-server-????????" 'Name=state,Values=available' \
		--query 'reverse(sort_by(Images, &CreationDate))[:1].ImageId' \
```

Then if you see any `${MODEL_ARCH:-}`, can you change it to `${MODEL_ARCH}`.

We shouldn't have optional globals everywhere.

Reply:
Done